enterprise/neo4j-enterprise/src/test/java/org/neo4j/kernel/builtinprocs/EnterpriseCreateIndexProcedureIT.java

```diff
@@ -0,0 +1,433 @@
+import static org.junit.Assume.assumeThat;
+import static org.neo4j.graphdb.factory.GraphDatabaseSettings.SchemaIndex.NATIVE10;
+import static org.neo4j.values.storable.Values.stringValue;
+
+@RunWith( Parameterized.class )
+public class EnterpriseCreateIndexProcedureIT extends KernelIntegrationTest
+{
+    @Parameterized.Parameters( name = "{2}" )
+    public static Collection<Object[]> parameters()
+    {
+        return Arrays.asList(
+                new Object[]{false, false, "createIndex", "index created"},
+                new Object[]{false, true, "createUniquePropertyConstraint", "uniqueness constraint online"},
+                new Object[]{true, true, "createNodeKey", "node key constraint online"}
+        );
+    }
+
+    @Parameterized.Parameter()
+    public static boolean existenceConstraint;
+
+    @Parameterized.Parameter( 1 )
+    public static boolean uniquenessConstraint;
+
+    @Parameterized.Parameter( 2 )
+    public static String indexProcedureName;
+
+    @Parameterized.Parameter( 3 )
+    public static String expectedSuccessfulCreationStatus;
+
+    @Test
+    public void createIndexWithGivenProvider() throws KernelException
+    {
+        testCreateIndexWithGivenProvider( "Person", "name" );
+    }
+
+    @Test
+    public void createIndexWithGivenProviderComposite() throws KernelException
+    {
+        testCreateIndexWithGivenProvider( "NinjaTurtle", "favoritePizza", "favoriteBrother" );
+    }
+
+    @Test
+    public void shouldCreateNonExistingLabelAndPropertyToken() throws Exception
+    {
+        // given
+        String label = "MyLabel";
+        String propKey = "myKey";
+        Transaction transaction = newTransaction( AnonymousContext.read() );
+        assertEquals( "label token should not exist", TokenRead.NO_TOKEN, transaction.tokenRead().nodeLabel( label ) );
+        assertEquals( "property token should not exist", TokenRead.NO_TOKEN, transaction.tokenRead().propertyKey( propKey ) );
+        commit();
+
+        // when
+        newTransaction( AnonymousContext.full() );
+        callIndexProcedure( indexPattern( label, propKey ), GraphDatabaseSettings.SchemaIndex.NATIVE20.providerName() );
+        commit();
+
+        // then
+        transaction = newTransaction( AnonymousContext.read() );
+        assertNotEquals( "label token should exist", TokenRead.NO_TOKEN, transaction.tokenRead().nodeLabel( label ) );
+        assertNotEquals( "property token should exist", TokenRead.NO_TOKEN, transaction.tokenRead().propertyKey( propKey ) );
+    }
+
+    @Test
+    public void throwIfNullProvider() throws Exception
+    {
+        // given
+        Transaction transaction = newTransaction( AnonymousContext.writeToken() );
+        transaction.tokenWrite().labelGetOrCreateForName( "Person" );
+        createProperties( transaction, "name" );
+        commit();
+
+        // when
+        newTransaction( AnonymousContext.full() );
+        String pattern = indexPattern( "Person", "name" );
+        try
```

Comment:
There's a better way to do this: `E e = Assertions.assertThrows(E.class, () -> {...})`